src/Tokenizer/Analyzer/FunctionsAnalyzer.php

```diff
@@ -33,8 +34,10 @@ final class FunctionsAnalyzer
 
     /**
      * Important: risky because of the limited (file) scope of the tool.
+     *
+     * @param list<NamespaceAnalysis> $namespaceDeclarations
      */
-    public function isGlobalFunctionCall(Tokens $tokens, int $index): bool
+    public function isGlobalFunctionCall(Tokens $tokens, int $index, array $namespaceDeclarations = null): bool
```

Comment:
Can we instead of passing `$namespaceDeclarations` here keep it in the `PhpCsFixer\Tokenizer\Tokens` class?

Adding `$namespaceDeclarations` would make it bug-prone as we would expect `$namespaceDeclarations` to be exactly namespace declarations for passed `$tokens`.

In  `PhpCsFixer\Tokenizer\Tokens` we could have it lazy loaded:
```php
<?php
class Tokens
{
    // ...
    private ?array $namespaceDeclarations = null;
    // ...
    public function getNamespaceDeclarations()
    {
        if (null === $this->namespaceDeclarations) {
            $this->namespaceDeclarations = (new NamespacesAnalyzer())->getDeclarations($tokens);
        }
    return $this->namespaceDeclarations;
    // ...
}

Reply:
Duplicate of [my comment](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6939#discussion_r1186047362), I'll resolve it to focus on one thread 😉.

Reply:
You were faster by 11 minutes, I did not see your comment yet when I wrote this one 😄